src/builder-manifest.c

```diff
@@ -2773,17 +2767,56 @@ builder_manifest_cleanup (BuilderManifest *self,
 
       if (self->appstream_compose && appdata_file != NULL)
         {
-          g_autofree char *basename_arg = g_strdup_printf ("--basename=%s", self->id);
-          g_print ("Running appstream-compose\n");
-          if (!appstream_compose (app_dir, error,
-                                  self->build_runtime ?  "--prefix=/usr" : "--prefix=/app",
-                                  "--origin=flatpak",
-                                  basename_arg,
-                                  self->id,
-                                  NULL))
+          g_autofree char *origin = g_strdup_printf ("--origin=%s",builder_manifest_get_id (self));
+          g_autofree char *components_arg = g_strdup_printf ("--components=%s", self->id);
+          g_autofree char *app_root_path = g_file_get_path (app_root);
+          g_autofree char *result_root_arg = g_strdup_printf ("--result-root=%s", app_root_path);
+          g_autoptr(GFile) xml_dir = flatpak_build_file (app_root, "share/app-info/xmls", NULL);
+          g_autoptr(GFile) icon_out = flatpak_build_file (app_root, "share/app-info/icons/flatpak", NULL);
+          g_autofree char *data_dir = g_strdup_printf("--data-dir=%s",flatpak_file_get_path_cached(xml_dir));
+          g_autofree char *icon_dir = g_strdup_printf("--icons-dir=%s",flatpak_file_get_path_cached(icon_out));
+          g_autofree char *opt_mirror_screenshots_url = builder_context_get_opt_mirror_screenshots_url(context);
+          gboolean opt_export_only = builder_context_get_opt_export_only(context);
+
+          if ( opt_mirror_screenshots_url && !opt_export_only) 
```

Comment:
```suggestion
          if (opt_mirror_screenshots_url && !opt_export_only) 
```

Reply:
Hmm based on glancing at the previous behavior, I think *both* clauses of this should be behind `if (!opt_export_only)`, and then this becomes `if (opt_mirror_screenshots_url)`? Since otherwise, leaving off opt_export_only will still result in running `appstreamcli`, which I don't *think* is what we want?

Reply:
UPDATE: no I just can't read code, this checks out, it's just weird semantics to begin with.